test/e2e/apps/deployment.go

```diff
@@ -134,6 +141,238 @@ var _ = SIGDescribe("Deployment", func() {
+		framework.ExpectNoError(err, "failed to list Deployments")
+		foundDeployment := false
+		for _, deploymentItem := range deploymentsList.Items {
+			if deploymentItem.ObjectMeta.Name == testDeploymentName &&
+				deploymentItem.ObjectMeta.Namespace == testNamespaceName &&
+				deploymentItem.ObjectMeta.Labels["test-deployment-static"] == "true" &&
+				*deploymentItem.Spec.Replicas == testDeploymentMinimumReplicas &&
+				deploymentItem.Spec.Template.Spec.Containers[0].Image == testDeploymentPatchImage {
+				foundDeployment = true
+				break
+			}
+		}
+		framework.ExpectEqual(foundDeployment, true, "unable to find the Deployment in list", deploymentsList)
+
+		ginkgo.By("updating the DeploymentStatus")
+		testDeploymentUpdate := testDeployment
+		testDeploymentUpdate.ObjectMeta.Labels["test-deployment"] = "updated"
+		testDeploymentUpdate.Spec.Template.Spec.Containers[0].Image = testDeploymentUpdateImage
+		testDeploymentDefaultReplicasPointer := &testDeploymentDefaultReplicas
+		testDeploymentUpdate.Spec.Replicas = testDeploymentDefaultReplicasPointer
+		testDeploymentUpdateUnstructuredMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&testDeploymentUpdate)
+		framework.ExpectNoError(err, "failed to convert to unstructured")
+		testDeploymentUpdateUnstructured := unstructuredv1.Unstructured{
+			Object: testDeploymentUpdateUnstructuredMap,
+		}
+		// currently this hasn't been able to hit the endpoint replaceAppsV1NamespacedDeploymentStatus
+		_, err = dc.Resource(deploymentResource).Namespace(testNamespaceName).Update(context.TODO(), &testDeploymentUpdateUnstructured, metav1.UpdateOptions{}) //, "status")
+		framework.ExpectNoError(err, "failed to update the DeploymentStatus")
+		foundEvent = false
+		for event := range dplmtWatchChan {
+			if event.Type == watch.Modified {
+				foundEvent = true
+				break
+			}
+		}
+		framework.ExpectEqual(foundEvent, true, "failed to find watch event %v", watch.Modified)
+
+		ginkgo.By("fetching the DeploymentStatus")
+		deploymentGetUnstructured, err := dc.Resource(deploymentResource).Namespace(testNamespaceName).Get(context.TODO(), testDeploymentName, metav1.GetOptions{}, "status")
+		framework.ExpectNoError(err, "failed to fetch the Deployment")
+		deploymentGet := appsv1.Deployment{}
+		err = runtime.DefaultUnstructuredConverter.FromUnstructured(deploymentGetUnstructured.Object, &deploymentGet)
+		framework.ExpectNoError(err, "failed to convert the unstructured response to a Deployment")
+		framework.ExpectEqual(deploymentGet.Spec.Template.Spec.Containers[0].Image, testDeploymentUpdateImage, "failed to update image")
+		framework.ExpectEqual(deploymentGet.ObjectMeta.Labels["test-deployment"], "updated", "failed to update labels")
+		foundEvent = false
+		for event := range dplmtWatchChan {
+			if event.Type == watch.Modified {
+				foundEvent = true
+				break
+			}
+		}
+		framework.ExpectEqual(foundEvent, true, "failed to find watch event %v", watch.Modified)
```

Comment:
What are we waiting for here exactly?

Reply:
The pattern that this test follows is, to after each modification, it will be verified that it has been modified.
This is useful for making sure there aren't conflicts with things like updates.